src/Features/CSharp/Portable/IntroduceVariable/CSharpIntroduceParameterService.cs

```diff
@@ -0,0 +1,82 @@
+        public CSharpIntroduceParameterService()
+        {
+        }
+
+        protected override SeparatedSyntaxList<SyntaxNode> AddArgumentToArgumentList(SeparatedSyntaxList<SyntaxNode> invocationArguments,
+                                                                                     SyntaxNode newArgumentExpression,
+                                                                                     int insertionIndex,
+                                                                                     string name,
+                                                                                     bool named)
+        {
+            ArgumentSyntax argument;
+            if (named)
+            {
+                var nameColon = SyntaxFactory.NameColon(name);
+                argument = SyntaxFactory.Argument(nameColon, default, (ExpressionSyntax)newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
+            }
+            else
+            {
+                argument = SyntaxFactory.Argument((ExpressionSyntax)newArgumentExpression.WithAdditionalAnnotations(Simplifier.Annotation));
+            }
+            return invocationArguments.Insert(insertionIndex, argument);
+        }
+
+        protected override SyntaxNode GenerateExpressionFromOptionalParameter(IParameterSymbol parameterSymbol)
+        {
+            return ExpressionGenerator.GenerateExpression(parameterSymbol.Type, parameterSymbol.ExplicitDefaultValue, canUseFieldReference: true);
+        }
+
+        protected override ImmutableArray<SyntaxNode> AddExpressionArgumentToArgumentList(ImmutableArray<SyntaxNode> arguments, SyntaxNode expression)
+        {
+            var newArgument = SyntaxFactory.Argument((ExpressionSyntax)expression);
+            return arguments.Add(newArgument);
+        }
+
+        protected override List<IParameterSymbol> GetParameterList(SemanticDocument document, SyntaxNode parameterList, CancellationToken cancellationToken)
+        {
+            var semanticModel = document.SemanticModel;
+            var parameterSyntaxList = ((ParameterListSyntax)parameterList).Parameters;
+            var parameterSymbolList = new List<IParameterSymbol>();
+
+            foreach (var parameterSyntax in parameterSyntaxList)
+            {
+                var symbolInfo = semanticModel.GetDeclaredSymbol(parameterSyntax, cancellationToken);
+                if (symbolInfo is IParameterSymbol parameterSymbol)
+                {
+                    parameterSymbolList.Add(parameterSymbol);
+                }
+            }
+
+            return parameterSymbolList;
+        }
+
+        protected override bool IsMethodDeclaration(SyntaxNode node)
+            => node.IsKind(SyntaxKind.LocalFunctionStatement) || node.IsKind(SyntaxKind.MethodDeclaration) || node.IsKind(SyntaxKind.SimpleLambdaExpression);
```

Comment:
Does ParameterizedLambdaExpression need to be here too?

Reply:
Probably, if someone can point me to a place where I can see all the different SyntaxNodes that have ParameterLists, I would probably need to use that.

Reply:
Yeah, I don't know of a nice list, I usually just play around in [sharplab.io](https://sharplab.io/#v2:C4LghgzsA0AmIGoA+ABATARgLACgUGYACdQgYUIG9dCbiiUAWQgWQAoBKa2qnWvwgG5gAToTCEAvIQAekgHyEMhBITQBuLv0EjCAI0mFW09vMXLVG3lu2iAxgY6mlK9ZsIBfXO6A) and look at the syntax tree and see what happens. I'm mainly commenting from experience doing edit and continue stuff, that every time I needed one type of lambda, I always needed the other :)

Reply:
won't work for constructors, or indexers.